### PR TITLE
Reverts 93bee53 and associated changes, adds observers to the respawn controller on RP mode

### PR DESCRIPTION
**Pull request by DisturbHerb**
[INPUT][RP][FEATURE]

## About the PR 
This PR may be controversial, please allow adequate time for feedback. This PR also does not in any way constitute an endorsement or repudiation of the current or past state of affairs regarding observer respawns.

Reverts commit [93bee53](https://github.com/goonstation/goonstation/commit/93bee53c2243e935e00315162632f8d3922cb1cf) and the other changes made to `observer.dm` to allow the game to compile properly. Choosing to observe the round will allow respawns after the standard respawn time when `RP_MODE` is true.

When choosing to observe from the splash screen, the pop-up will inform the player that they have the option to participate when their respawn timer runs out.
An image of the observe pop-up box on the RP servers.

## Why's this needed? 
For a long time, the continued inclusion of this feature/bug didn't seem to garner a consensus among the development team and was generally left alone until #12750 prompted a change. This change has been met with some level of controversy, and this PR attempts to present another option for retaining that feature while also making it explicit that it _is_ a feature.

On RP, ghosts who have participated in the round and either die or cryo are already permitted to respawn after a timer. It may be incongruent to disallow respawns from ghosts who choose to observe the round from the outset, and the rules are already explicit on respawning players and metagaming.

## Changelog 

```changelog
(u)DisturbHerb
(*)Observer respawn has been re-implemented for the roleplay servers. The pop-up will explicitly inform you that you are unable to participate until the respawn timer runs out. Please do not use this to metagame.
```

### CODE REVIEW

**Comment by TheTyrant25** [comment]
It makes sense for it to be the same.

There have been issues with respawning once DNR was set and whether that's how it should work but that's out of scope here I think

**Comment by Azrun** [comment]
> This PR also does not in any way constitute an endorsement or repudiation of the current or past state of affairs regarding observer respawns.

Isn't that what a PR is though? You are suggesting a change be made.

> This change has been met with some level of controversy, and this PR attempts to present another option for retaining that feature while also making it explicit that it is a feature.

Are you saying that you personally don't agree with the outcomes proposed by this PR but it captures what other people want/desire? Helps with clarifying if you would be speaking to the intent of the change vs only the implementation.

File: code/mob/dead/observer.dm [hunk 1]
```diff
@@ -331,8 +330,7 @@
 		if(istype(get_area(src),/area/afterlife))
 			qdel(src)
 
-		if(!istype(src, /mob/dead) && !mind?.dnr)
-			respawn_controller.subscribeNewRespawnee(our_ghost.ckey)
+		respawn_controller.subscribeNewRespawnee(our_ghost.ckey)
```
Comment (ZeWaka) [inline code/mob/dead/observer.dm:333]: this would need a comment then to make it clear

**Review by ZeWaka — COMMENTED** [review]
Design Opinion: tbh i don't think this should differ between rp and non-rp, however this ends up

**Comment by DisturbHerb** [comment]
> Are you saying that you personally don't agree with the outcomes proposed by this PR but it captures what other people want/desire?

i didn't quite think that my opinions on the matter were all that important, i just didn't want to influence the discussion on this issue too much. i'm just the messenger 👻

**Pull request closed without merging**